### PR TITLE
WIP neux lessons in the app router

### PR DESCRIPTION
**Pull request by joelhooks**

### CODE REVIEW

File: src/app/(content)/courses/[course]/[lesson]/page.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import {loadLesson} from '@/lib/lessons'
+import {loadCourse} from '@/lib/courses'
+import {LessonResource} from '@/types'
+import {notFound} from 'next/navigation'
+import {PlayerTwo} from '@/app/(content)/courses/[course]/[lesson]/Player'
```
Comment (zacjones93) [inline src/app/(content)/courses/[course]/[lesson]/page.tsx:5]: hadn't seen the `(content)` route grouping before but I like it

https://nextjs.org/docs/app/building-your-application/routing/route-groups#examples

File: next.config.js [hunk 2]
```diff
@@ -41,6 +41,9 @@ const IMAGE_HOST_DOMAINS = compact([
 ])
 
 const nextConfig = {
+  experimental: {
+    ppr: true,
```
Comment (zacjones93) [inline next.config.js:45]: partial prerendering, fancy https://nextjs.org/docs/app/api-reference/next-config-js/partial-prerendering

File: src/app/(content)/courses/[course]/[lesson]/page.tsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+const LessonHeader = async ({
+  lessonLoader,
+}: {
+  lessonLoader: Promise<LessonResource>
+}) => {
+  const lesson = await lessonLoader
+
+  if (!lesson) {
+    return notFound()
```
Comment (zacjones93) [inline src/app/(content)/courses/[course]/[lesson]/page.tsx:58]: haven't seen this function, that's handy https://nextjs.org/docs/app/api-reference/functions/not-found

File: src/pages/courses/[course]/index.tsx [hunk 4]
```diff
@@ -120,11 +120,11 @@ export const getServerSideProps: GetServerSideProps = async ({
   query,
 }) => {
   setupHttpTracing({name: getServerSideProps.name, tracer, req, res})
-
+  debugger
```
Comment (zacjones93) [inline src/pages/courses/[course]/index.tsx:123]:
```suggestion
```
probably don't want debugger statement here?

File: src/components/search/pagination.tsx [hunk 5]
```diff
@@ -10,21 +10,10 @@ const Pagination = () => {
...
-          <svg className="my-1" xmlns="http://www.w3.org/2000/svg" width="16" height="16" viewBox="0 0 16 16"><g fill="currentColor" fillRule="evenodd" transform="translate(3 3)"><path d="M0.293,9.707 C-0.0973819383,9.31650015 -0.0973819383,8.68349985 0.293,8.293 L3.586,5 L0.293,1.707 C-0.0859722095,1.31462111 -0.0805523739,0.690915223 0.305181425,0.305181425 C0.690915223,-0.0805523739 1.31462111,-0.0859722095 1.707,0.293 L5.707,4.293 C6.09738194,4.68349985 6.09738194,5.31650015 5.707,5.707 L1.707,9.707 C1.31650015,10.0973819 0.683499851,10.0973819 0.293,9.707 Z"/><path d="M5.293,9.707 C4.90261806,9.31650015 4.90261806,8.68349985 5.293,8.293 L8.586,5 L5.293,1.707 C4.91402779,1.31462111 4.91944763,0.690915223 5.30518142,0.305181425 C5.69091522,-0.0805523739 6.31462111,-0.0859722095 6.707,0.293 L10.707,4.293 C11.0973819,4.68349985 11.0973819,5.31650015 10.707,5.707 L6.707,9.707 C6.31650015,10.0973819 5.68349985,10.0973819 5.293,9.707 Z"/></g></svg>,
-        page(currentRefinement: any) {
-          return <div className="px-1">{currentRefinement}</div>
-        },
+        firstPageItemText: 'First',
+        previousPageItemText: 'Previous',
+        nextPageItemText: 'Next',
```
Comment (Creeland) [inline src/components/search/pagination.tsx:15]: Replaced the SVGs with text

File: src/components/search/stats.tsx [hunk 6]
```diff
@@ -1,28 +1,23 @@
...
+import {useStats, StatsProps} from 'react-instantsearch'
 
-type CustomStatsProps = {
-  nbHits: number
-  searchQuery: string
-}
+const Stats: React.FunctionComponent<React.PropsWithChildren<StatsProps>> = (
+  props,
+) => {
+  const {nbHits, query} = useStats(props)
```
Comment (Creeland) [inline src/components/search/stats.tsx:7]: The code mod script initially created custom connect functions that use the hooks API.

I decided to instead integrate the hooks pattern into the components.

**Review by Creeland — APPROVED** [review]

**Pull request merged**